### ISSUE TITLE
GPT-2 doesn't always have an `lm_head`

### ISSUE THREAD

**Issue by steventrouble** [rustformers/llm#338]
The GPT2 ggml models don't always have an lm_head, but `llm` assumes it does:

https://github.com/rustformers/llm/blob/9a222690cd0a9e3322bb6a926d54e90d08fb2c0f/crates/models/gpt2/src/lib.rs#L62

See ggml example repo where they mention lm_head is optional: https://github.com/ggerganov/ggml/blob/ee1b3727e60403012dd2b57d35b60558f4db66d8/examples/gpt-2/main.cpp#L364

**Comment by steventrouble** [comment]
I tried replacing it with

```rust
let lm_head = tl.load("model/lm_head").or_else(|_| tl.load("model/wte"))?;
```

But then I ran into a more ggml errors and had to move on

```
ggml_new_tensor_impl: not enough space in the context's memory pool (needed 252199424, available 250733056)
```

**Comment by LLukas22** [comment]
Yeah i also noticed that, we probably have to share the wte tensor as the context size is calculated via the combined size of all tensors.
